docs/testing_using_Chai_&_Mocha.md

```diff
@@ -0,0 +1,161 @@
+Testing using Chai & Mocha
+===================
+
+_(Supported since Remix IDE v0.22.0)_
+
+Remix supports testing of your files in JavaScript using assertion library [Chai](https://www.chaijs.com/) & test framework [Mocha](https://mochajs.org/)
+
+_Chai is a BDD / TDD assertion library for node and the browser that can be delightfully paired with any javascript testing framework._
+
+_Mocha is a feature-rich JavaScript test framework running on Node.js and in the browser, making asynchronous testing simple and fun._
+
+## Write tests
+
+Create a js file in your project workspace. Better to create it inside `scripts` folder. Lets name it `sample.test.js`.
+
+Write your tests in the file. Here is a sample:
+
+```
+const { expect } = require("chai");
+
+describe("Sample", function () {
+  it("Sample tests with mocha and chai", async function () {
+    var foo = 'bar'
+    var beverages = { tea: [ 'chai', 'matcha', 'oolong' ] };
+    expect(foo).to.be.a('string');
+    expect(foo).to.equal('bar');
+    expect(foo).to.have.lengthOf(3);
+    expect(beverages).to.have.property('tea').with.lengthOf(3);
+  });
+});
+```
+
+## Run tests
+
+Once done with writing the tests, right click on file name in `File Explorers` plugin. It will show some options along with option to `Run with Mocha`.
+
+![](images/run_with_mocha_option.png)
+
+Click on `Run with Mocha` option, tests will be executed and result will be shown on Terminal.
+
+![](images/run_with_mocha_result.png)
+
+## Test a contract
+
+Similarly unit tests can be written to test the functionality of a smart contract. An example to test default `1_Storage.sol` contract can be as:
```

Comment:
Maybe after the 1st sentence, add a sentence like:
The tests are run on the artifact file (the .json file) that is generated when the .sol file has been compiled.

Reply:
added something similar in the code itself